DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py

```diff
@@ -0,0 +1,137 @@
+         5.5 Verify traffic with vlan "4" received on port 4
+    """
+
+    # 1Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    dent_dev = dent_devices[0]
+    device = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    ports = tgen_dev.links_dict[device][1]
+    default_pvid = 1
+    tx_port = 0
+
+    out = await IpLink.add(input_data=[{device: [{"device": "br0", "type": "bridge", "vlan_filtering": 1}]}])
+    assert out[0][device]["rc"] == 0, f"Failed creating bridge.\n {out}"
+
+    await IpLink.set(input_data=[{device: [{"device": "br0", "operstate": "up"}]}])
+    assert out[0][device]["rc"] == 0, f"Failed setting bridge to state UP.\n {out}"
+
+    out = await IpLink.set(input_data=[{device: [{"device": port, "operstate": "up"} for port in ports]}])
+    assert out[0][device]["rc"] == 0, f"Failed setting link to state UP\n {out}"
+
+    await IpLink.set(input_data=[{device: [{"device": port, "master": "br0"} for port in ports]}])
+    assert out[0][device]["rc"] == 0, f"Failed to enslave link(s) to bridge.\n {out}"
+
+    # 2.Set links to vlans.
+    for config in port_map:
+        port = ports[config["port"]]
+        out = await BridgeVlan.add(input_data=[{device: [{"device": port, "vid": setting["vlan"],
+                                                          "pvid": setting["pvid"], "untagged": setting["untagged"]} for
+                                                         setting in config["settings"]]}])
+        assert out[0][device]["rc"] == 0, f'Vlan: {config["settings"]["pvid"]}  is not added to link: {port}\n {out}'
+
+    # 3. Map receiving and non receiving ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[3]][0]["name"], dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {}
+    for vlan in packet_vids:
+        vlan = vlan if vlan != "X" else ""
+
+        streams.update({f"traffic with VLAN ID: {vlan}": {
+            "type": "raw",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "srcMac": "02:00:00:00:00:AA",
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "vlanID": vlan,
+
+        }})
```

Comment:
You can use a dict comprehension here instead of a for loop:
```python
streams = {
    f"traffic with VLAN ID: {vlan}": {
            "type": "raw",
            "ip_source": tx_ports,
            "ip_destination": rx_ports,
            "srcMac": "02:00:00:00:00:AA",
            "dstMac": "ff:ff:ff:ff:ff:ff",
            "vlanID": vlan if vlan != "X" else None,
    } for vlan in packet_vids
}
```